django/shortcuts.py

```diff
@@ -41,11 +43,19 @@ def redirect(to, *args, permanent=False, **kwargs):
         * A URL, which will be used as-is for the redirect location.
 
     Issues a temporary redirect by default; pass permanent=True to issue a
-    permanent redirect.
+    permanent redirect. Pass preserve_method=True to issue a redirect tha must
+    preserve HTTP verb.
     """
-    redirect_class = (
-        HttpResponsePermanentRedirect if permanent else HttpResponseRedirect
-    )
+
+    redirect_class = HttpResponseRedirect
+
+    if not preserve_method and permanent:
+        redirect_class = HttpResponsePermanentRedirect
+    if preserve_method and not permanent:
+        redirect_class = HttpResponseTemporaryRedirectWithSameMethod
+    if preserve_method and not permanent:
+        redirect_class = HttpResponsePermanentRedirectWithSameMethod
```

Comment:
The 2nd and 3rd if statements have the same conditional. Should the 3rd be `preserve_method and permanent`? Would it be better to use if-else statements?

```suggestion
    if not preserve_method and permanent:
        redirect_class = HttpResponsePermanentRedirect
    elif preserve_method and not permanent:
        redirect_class = HttpResponseTemporaryRedirectWithSameMethod
    elif preserve_method and permanent:
        redirect_class = HttpResponsePermanentRedirectWithSameMethod
```